src/uwtools/config/core.py

```diff
@@ -568,25 +570,25 @@ def _yaml_loader(self) -> type[yaml.SafeLoader]:
 
     # Public methods
 
-    def dump_file(self, path: str) -> None:
+    @staticmethod
+    def dump_dict(path: OptionalPath, cfg: dict, opts: Optional[ns] = None) -> None:
         """
-        Dumps the config as a YAML file.
+        Dumps a provided config dictionary as a YAML file.
 
         :param path: Path to dump config to.
+        :param cfg: The in-memory config object to dump.
+        :param opts: Other options required by a subclass.
         """
-        YAMLConfig.dump_file_from_dict(path, self.data)
+        with writable(path) as f:
+            yaml.dump(cfg, f, sort_keys=False)
 
-    @staticmethod
-    def dump_file_from_dict(path: str, cfg: dict, opts: Optional[ns] = None) -> None:
+    def dump_file(self, path: DefinitePath) -> None:
         """
-        Dumps a provided config dictionary as a YAML file.
+        Dumps the config as a YAML file.
 
         :param path: Path to dump config to.
-        :param cfg: The in-memory config object to dump.
-        :param opts: Other options required by a subclass.
         """
-        with open(path, "w", encoding="utf-8") as file_name:
-            yaml.dump(cfg, file_name, sort_keys=False)
+        YAMLConfig.dump_dict(path, self.data)
 
 
 class FieldTableConfig(YAMLConfig):
```

Comment:
Must appear out-of-order after `YAMLConfig` since it refers to that class as a parent class.